src/main/java/seedu/address/model/tag/TagSet.java

```diff
@@ -0,0 +1,30 @@
+package seedu.address.model.tag;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class TagSet {
+    /**
+     * Combines the results of removing {@code tagsToDelete} from {@code originalTags} adding {@code tagsToAdd}.
+     * @param originalTags Original collection of tags.
+     * @param tagsToAdd A collection of tags to be added.
+     * @param tagsToDelete A collection of tags to be removed from {@code originalTags}.
+     * @return
+     */
+    public static Set<Tag> mergeAndRemove(Collection<Tag> originalTags, Collection<Tag> tagsToAdd,
```

Comment:
I'm curious, why did you abstract this method out into its own class?

Reply:
I was debating between writing mergeAndRemove within EditCommand and creating a new TagSet class that actually encapsulates a set with additional functionalities. Hence, I decided on the middle ground of creating a helper class with static methods instead to manipulate a set of tags. In hindset, in order to produce `CommandResult` responsively, it is actually easier to put this inside the EditCommand class.